### PR TITLE
refactor(examples): Better devMode usage

### PR DESCRIPTION
**Pull request by obartra**
- Refactor so usage examples need less custom css
- Ensure devMode is used by all examples

Closes https://github.com/obartra/reflex/issues/174

### CODE REVIEW

File: storybook/shared/box.js [hunk 1]
```diff
@@ -8,39 +8,34 @@ const typeMap = {
...
-          {children || value}
-        </Grid>
+export default function Box({
+  children,
+  margin = [0, 0.5, 1],
+  style,
+  type,
+  value = type,
+  ...props
+}: {
```
Comment (arahansen) [inline storybook/shared/box.js]: This might be good to pull outside the function parameters. A bit hard to parse when its inlined like this.

File: storybook/stories/examples/cards.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React from 'react'
 import { Grid, Layout, Root } from 'reflex'
 import { Box } from '../../shared'
 import styles from './layout.css'
-import { top, horizontalHalf } from '../../shared/marginTypes'
```
Comment (arahansen) [inline storybook/stories/examples/cards.js]: hmm why did the screenshot pic so subtly change here? is that significant enough to look into?

Comment (obartra) [inline storybook/stories/examples/cards.js]: it's intentional. Things were relying on hardcoded margins and I replaced all these by actual reflex margin properties. So they don't match exactly but I think this is a better use of them.

E.g. in one instance `h1` had a `10px 0` margin so I replaced that with `verticalHalf` which is `12px` instead of `10px`
